README.md

```diff
@@ -78,14 +101,97 @@ Rather it provides a solution that solves several problems attributable to the c
+
+You don't necessarily need Redux-cycle if your goal is only that.
+You might find Redux-saga to be easier to switch to.
+
+### I already know Redux-saga
+
+Redux-cycles can help you to:
+
+* Handle your side-effects declaratively.
+
+  Side-effect handling in Redux-saga makes testing easier compared to thunks, but you're still ultimately doing glorified function calls.
+  The Cycle.js architecture pushes side-effect handling further to the edges of your application, leaving your "cycles" (or "epcis") operate on pure streams.
+
+* Type your business logic.
+
+  Most of your business logic lives in sagas... and they are hard to impossible to statically type.
+  Have you had silly bugs in your sagas that Flow could have caught?
+  I sure had.
+
+### I already know Redux-observable
+
+Redux-cycles appears to be similar to Redux-observable... which it is, due to embracing observables.
+So why might you want to try Redux-cycles?
+
+In a word: easier side-effect handling.
+With Redux-observable your side-effectful code is scattered through all your epics, *directly*.
+
+It's hard to test.
+The code is less legible.
+
+### Do I have to buy all-in?
+
+Should you go ahead and rewrite the entirety of your application in Redux-cycles to take advantage of it?
+
+**Not at all.**
+
+It's not the best strategy really.
+What you might want to do instead is to identify a small distinct "category" of side-effectful logic in your current side-effect model, and try transitioning only this part to use Redux-cycles, and see how you feel.
+
+A great example of a small category like that could be:
+
+* local storage calls
+* payments API
+
+The domain API layer often is not the easiest one to switch, so if you're thinking that... think of something smaller :)
+
+**Redux-saga** can still be valuable, even if using Redux-cycles.
+Certain sagas read crystall clear, sagas that orchestrate user flow.
+
+Like onboarding maybe: after the user signs up, and adds two todos, show a "keep going!" popup.
+
+This kind of logic fits the imperative sagas model *perfectly*, and it will likely look more cryptic if you try to redo it reactively.
+
+Life's not all-or-nothing, you can definitely use Redux-cycles and Redux-saga side-by-side.
+
 ## What's this Cycle thing anyway?
 
 [Cycle.js](https://cycle.js.org) is an interesting and unusual way of represting real-world programs.
 
 The program is represented as a pure function, which takes in some *sources* about events in the real world (think a stream of Redux actions), does something with it, and returns *sinks*, aka streams with commands to be performed.
 
+<dl>
+  <dt><strong>stream</strong></dt>
+  <dd>is like an asynchronous, always-changing array of values</dd>
+  <dt><strong>source</strong></dt>
+  <dd>is a stream of real-world events as they happen</dd>
+  <dt><strong>sink</strong></dt>
+  <dd>is a stream of commands to be performed</dd>
+  <dt><strong>cycle/epic</strong></dt>
```

Comment:
since this section is about Cycle.js perhaps we should have "epic" here as it's confusing (doesn't take sources/return sinks).